wrappers/python/CMakeLists.txt

```diff
@@ -24,14 +24,31 @@ if (NOT hasParent)
     if (NOT BUILD_SHARED_LIBS)
         set(CMAKE_POSITION_INDEPENDENT_CODE ON)
     endif()
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../../core ZXing EXCLUDE_FROM_ALL)
-
-    enable_testing()
+    if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../../core)
+        add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../../core ZXing EXCLUDE_FROM_ALL)
+    else()
+        find_package (Git REQUIRED)
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} ls-remote https://github.com/nu-book/zxing-cpp.git tags/*
+            RESULT_VARIABLE RESULT
+            OUTPUT_VARIABLE OUTPUT)
+        string (FIND "${OUTPUT}" "refs/tags/v${VERSION_INFO}\n" FOUND_IDX)
+        if (FOUND_IDX LESS 0)
+            # Version not found in tags (suppose we are preparing future version), use master branch
+            FetchContent_Declare(zxing-cpp
+                GIT_REPOSITORY https://github.com/nu-book/zxing-cpp.git)
+        else()
+            FetchContent_Declare(zxing-cpp
+                GIT_REPOSITORY https://github.com/nu-book/zxing-cpp.git
+                GIT_TAG v${VERSION_INFO})
+        endif()
+        if(NOT zxing-cpp_POPULATED)
+            FetchContent_Populate(zxing-cpp)
+            add_subdirectory(${zxing-cpp_SOURCE_DIR}/core ZXing EXCLUDE_FROM_ALL)
+        endif()
+    endif()
 endif()
 
 # build the python module
 pybind11_add_module(zxing zxing.cpp)
 target_link_libraries(zxing PRIVATE ZXing::ZXing)
-
```

Comment:
Ok. I reset it but changed the test launch command : using "python setup.py test" do install dependencies (numpy) so that ctest does not fail and do really execute some of the test instead of skipping tests requiring numpy.

Reply:
Please revert that just as is was, which means also revert the disabling of the `@unittest.skipIf(not has_numpy` checks in `test.py`. Everything gets fully tested in your new CI scripts anyway.

Reply:
Done. I just split the unittests in two classes so that we can set the @unittest.skipIf(not has_numpy check on the whole class instead of repeating for each test